src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -334,16 +334,16 @@ public static void WriteTo(TextWriter writer, HtmlEncoder encoder, object value)
             if (htmlContent != null)
             {
                 var htmlTextWriter = writer as HtmlTextWriter;
-                if (htmlTextWriter == null)
-                {
-                    htmlContent.WriteTo(writer, encoder);
-                }
-                else
+                if (htmlTextWriter != null)
                 {
                     // This special case allows us to keep buffering as IHtmlContent until we get to the 'final'
                     // TextWriter.
                     htmlTextWriter.Write(htmlContent);
                 }
+                else
+                {
+                    htmlContent.WriteTo(writer, encoder);
+                }
```

Comment:
Yup, exactly. I was looking at the code special-casing `HtmlTextWriter` and the inconsistencies started to get confusing. This also moves the `var` closer to its use. But I can revert if you dislike.